qiskit_aqua/utils/circuit_cache.py

```diff
@@ -0,0 +1,200 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018 IBM.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+""" A utility for caching and reparameterizing circuits, rather than compiling from scratch
```

Comment:
Does it work with any pass in transpiler? or it will be safer to use empty transpiler? Do you mind updating the note? Thanks.

Reply:
Good catch, updated language.